levelkdev/swapr-dapp
# Feature/906 add dfyn dex

# Summary

Fixes #906 
Adds DFYN to the routing options when swapping.

![image](https://user-images.githubusercontent.com/36545016/165067722-705abb15-a51f-4e17-b0a4-1e72e6340a75.png)
|:--:|
| <b>Image - On localhost</b>|

### Background
In order to add more options/value to our platform, we want to add more relevant dexes to the list of swap options.


  # To Test/Replicate

1. Follow the procedure to link the SDK locally (see below)
2. `yarn build`
3. Change network to `Polygon`
4. Select compatible token pair.


### IMPORTANT
This PR uses unreleased SDK.

TL;DR
To make it work you need to build & link [this SDK branch](https://github.com/levelkdev/swapr-sdk/tree/feature/add-polygon-config) locally.

How to do that: (from [relevant PR](https://github.com/levelkdev/swapr-dapp/pull/910))

Clone SDK repo locally
Checkout to `feature/add-polygon-config` on the SDK repo
run `yarn prepublishOnly`
run `yarn link`
cd to swapr-dApp and run `yarn link @swapr/sdk`
All types/missing imports should work fine now.

After you're done playing with linked repo you need to unlink local build. To do this, in your swapr-dApp `run yarn unlink @swapr/sdk && yarn install --force`
